appserver/web/war-util/src/main/java/org/glassfish/web/loader/WebappClassLoader.java

```diff
@@ -1188,6 +1189,17 @@ public String toString() {
 
     public void preDestroy() {
         LOG.log(TRACE, "preDestroy()");
+
+        // Close deployment time libraries.
+        ClassLoader parent = getParent();
```

Comment:
Could we extract new method, name it like `close deployment time libraries`, call it here, and avoid comment?

Reply:
Yeah, I support this idea. Are we sure that those parent classloaders are not shared?

I always avoided using external dependencies from the file system. I have doubts it always caused issues. It can be in the domain, but then it is manager by the domain cl, or in ear/war, then it is bound to the app lifecycle. Otherwise... issues.

Reply:
> Could we extract new method, name it like `close deployment time libraries`, call it here, and avoid comment?

Certainly. Thank you for advice.

Reply:
> I always avoided using external dependencies from the file system. I have doubts it always caused issues. It can be in the domain, but then it is manager by the domain cl, or in ear/war, then it is bound to the app lifecycle. Otherwise... issues.

This is an attempt to solve test issue on Windows (again ;( ). This line failed:

https://github.com/eclipse-ee4j/glassfish/blob/6a77e1cf5f69666837d487232b1c132c4f175d53/appserver/tests/application/src/test/java/org/glassfish/main/test/app/security/jmac/http/servlet/basic/HttpServletBasicAuthTest.java#L111

when trying to delete auth module file.